hack/generator/pkg/astmodel/type_definition.go

```diff
@@ -136,15 +136,15 @@ func FileNameHint(name TypeName) string {
 func (def TypeDefinition) ApplyObjectTransformation(transform func(*ObjectType) (Type, error)) (TypeDefinition, error) {
 	// We use a TypeVisitor to allow automatic handling of wrapper types (such as ArmType and StorageType)
 	visited := false
-	visitor := MakeTypeVisitor()
-	visitor.VisitObjectType = func(_ *TypeVisitor, ot *ObjectType, _ interface{}) (Type, error) {
-		rt, err := transform(ot)
-		if err != nil {
-			return nil, err
-		}
-		visited = true
-		return rt, nil
-	}
+	visitor := MakeTypeVisitor(
+		func(_ *TypeVisitor, ot *ObjectType, _ interface{}) (Type, error) {
+			rt, err := transform(ot)
+			if err != nil {
+				return nil, err
+			}
+			visited = true
+			return rt, nil
+		})
```

Comment:
This sorta seems _less_ clear than it was before (to me). Certainly if it's a clarity/simplicity improvement it's a minor one.

Do we really want to give up compile-time type safety of fields for what feels like a small net gain? (I suppose it's nice that the visitor isn't modified after creation but we didn't actually _forbid_ modification so it seems a small win).

Supporting this also results in additional somewhat involved code to do the reflection, as well as tests for that. Is that additional complexity (and risk, although it's been well mitigated with tests) worth it over the dumb but "obviously correct" way it was before?

Reply:
For example it seems like it would be easy to accidentally pass two functions that had the same signature and have the second overwrite the first, for example. Something that would be more difficult to do with the old property-assignment approach since `visitor.VisitTypeName = myVisitTypeName` where `myVisitTypeName` actually has an `ObjectType` signature won't compile. If you _did_ accidentally do:
```
visitor.VisitObjectType = myVisitObjectType
...
visitor.VisitObjectType = myVisitTypeName
```

it's at least clear-ish from a scan of the code that you've overwritten something.

Reply:
Resolved with restructure.